api/base/serializers.py

```diff
@@ -87,29 +96,51 @@ def _get_attr_from_tpl(attr_tpl, obj):
     else:
         return attr_tpl
 
+
 # TODO: Make this a Field that is usable on its own?
 class Link(object):
     """Link object to use in conjunction with Links field. Does reverse lookup of
     URLs given an endpoint name and attributed enclosed in `<>`.
     """
 
-    def __init__(self, endpoint, args=None, kwargs=None, **kw):
+    def __init__(self, endpoint, args=None, kwargs=None, query_kwargs=None, **kw):
         self.endpoint = endpoint
         self.kwargs = kwargs or {}
         self.args = args or tuple()
         self.reverse_kwargs = kw
+        self.query_kwargs = query_kwargs or {}
 
     def resolve_url(self, obj):
         kwarg_values = {key: _get_attr_from_tpl(attr_tpl, obj) for key, attr_tpl in self.kwargs.items()}
         arg_values = [_get_attr_from_tpl(attr_tpl, obj) for attr_tpl in self.args]
+        query_kwarg_values = {key: _get_attr_from_tpl(attr_tpl, obj) for key, attr_tpl in self.query_kwargs.items()}
+
         return absolute_reverse(
             self.endpoint,
             args=arg_values,
             kwargs=kwarg_values,
+            query_kwargs=query_kwarg_values,
             **self.reverse_kwargs
         )
 
 
+class WaterbutlerLink(object):
+    """Link object to use in conjunction with Links field. Builds a Waterbutler URL for files.
+    """
+
+    def __init__(self, args=None, kwargs=None, **kw):
```

Comment:
You could add `query_args`, like `Link` has, and pass those to `waterbutler_url_for`.